core/src/main/java/com/valtech/aem/saas/core/fulltextsearch/SearchResultsServlet.java

```diff
@@ -0,0 +1,82 @@
+package com.valtech.aem.saas.core.fulltextsearch;
+
+import com.day.cq.wcm.api.Page;
+import com.valtech.aem.saas.core.common.page.ContainingPage;
+import com.valtech.aem.saas.core.common.resource.ResourceChildren;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletResponse;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.request.RequestDispatcherOptions;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.servlets.HttpConstants;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.apache.sling.servlets.annotations.SlingServletResourceTypes;
+import org.osgi.service.component.annotations.Component;
+
+@Slf4j
+@Component(service = Servlet.class)
+@SlingServletResourceTypes(
+    resourceTypes = SearchResultsServlet.RESOURCE_TYPE_CQ_PAGE,
```

Comment:
the request should not be bound to the page but it should be bound to the search component. you could have as well several search components on the page (in theory)